wall_e/extensions/sfu.py

```diff
@@ -502,6 +506,104 @@ async def outline(self, ctx, *course):
         if e_obj is not False:
             await ctx.send(embed=e_obj, reference=ctx.message)
 
+    @app_commands.command(name="courses", description="Gets all offered courses")
+    @app_commands.describe(department="Specify the department to search. Examples: STAT, PHYS, MATH")
+    @app_commands.describe(level="Specifies the level of courses to filter for. Examples: 100, 200, 400")
+    @app_commands.describe(term="Specify the specific semester to search for. Examples: Spring, Summer, Fall")
+    @app_commands.describe(year="Specify the specific year to search for. Examples: 2020, 2024, 2025")
+    async def courses(self, interaction: discord.Interaction, department: str = "", level: int = 0,
+                      term: str = "registration", year: str = "registration"):
+        self.logger.info(
+            f"[SFU courses()] courses command detected from user {interaction.user} with arguments: "
+            f"department {department}, level {level}, term {term}, year {year}"
+        )
+        await interaction.response.defer()
+
+        # The default course selection if not specified
+        departments = ["CMPT", "MATH", "MACM"]
+        if department:
+            departments = [department.upper()]
+
+        if level != 0 and (level < 100 or level >= 1000):
+            self.logger.debug("[SFU courses()] incorrect level arguments, defaulting to 0")
+            level = 0
+
+        courses = []
+        for department in departments:
+            url = f"http://www.sfu.ca/bin/wcm/course-outlines?{year}/{term}/{department}/"
+            self.logger.debug(f"[SFU courses()] url for get constructed: {url}")
+
+            res = await self.req.get(url)
+            if res.status == 200:
+                self.logger.debug(f"[SFU courses()] get request for {department} successful, parsing data")
+                res_json = await res.json()
+
+                # parse data for displaying, sorting, and filtering
+                for course in res_json:
+                    course["text"] = f"{department}{course['text']}"
+
+                    # we assume all courses have 3 digits
+                    if len(course["value"]) == 4:
+                        course["value"] = course["value"][:3]
+
+                    course["value"] = int(course["value"])
```

Comment:
We can make this block a bit easier to read by making it a single assignment
```suggestion
                    # we assume all courses have 3 digits
                    course["value"] = int(course["value"][:3] if len(course["value"]) == 4 else course["value"])
```

Reply:
you should drop the if branch and just do the truncation, if the value is 3 digits the truncation won't do anything.
```python
a='105w'
b='105'
c='25'

a[:3] # 105
b[:3] # 105
c[:3] # 25
```

Reply:
ooh, good point. Smart!